File: javascript/leetcode/144. binary tree preorder traversal_iterative.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var preorderTraversal = function(root) {
  let array = new Array();

  if (root === null) return array;

  let stack = new Array();
  stack.push(root);

  while (stack.length != 0) {
    let top = stack.pop();
    array.push(top.val);

    if (top.right) stack.push(top.right);
    if (top.left) stack.push(top.left);
  }
  return array;
};

File: python/leetcode/144. binary tree preorder traversal_iterative.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def preorderTraversal(self, root: TreeNode) -> List[int]:
        array = []
        
        if root is None:
            return array
        stack = []
        stack.append(root) 
        
        while len(stack) > 0:
            top = stack.pop()
            array.append(top.val)
            
            if top.right:
                stack.append(top.right)
            if top.left:
                stack.append(top.left)
        
        return array
        